02_bubble/vinilapeddireddy/bubblesort.go

```diff
@@ -0,0 +1,28 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+// returns a sorted copy of int slice items using Bubble sort:
+func bubble(items []int) []int {
+
+	n := len(items)
+
+	for i := 0; i < n; i++ {
+		for j := n - 1; j >= i+1; j-- {
+			if items[j] < items[j-1] {
+				items[j], items[j-1] = items[j-1], items[j]
+			}
+		}
+	}
+	return items
```

Comment:
Can't mutate the original. Make a copy. But first write a test that detects the lack of copying.

Reply:
Done